### PR TITLE
Distinct subsets of a set [Backtracking] C++

### PR DESCRIPTION
**Pull request by somya51p**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

From a set of positive integers, finding all its subsets. The set can contain duplicate elements, so any repeated subset should be considered only once. Used the concept of Backtracking to built the algorithm.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

This pull request is made for issue number #3827

### CODE REVIEW

**Comment by somya51p** [comment]
@iamrajiv Please review the Pull request.

File: C-Plus-Plus/other/distinct_subsets_of_set.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+    vector<int> arr;
+    int n = arr.size();
+
+    printPowerSet(arr, n);
+
+    return 0;
+}
+
+// Input: 10 12 12
```
Comment (iamrajiv) [inline C-Plus-Plus/other/distinct_subsets_of_set.cpp]: Change all single-line comment in one multiline comment.

File: C-Plus-Plus/other/distinct_subsets_of_set.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
+//Finding all distinct subsets of a given set using BitMasking Approach
```
Comment (iamrajiv) [inline C-Plus-Plus/other/distinct_subsets_of_set.cpp]: Change all single-line comment in one multiline comment.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@somya51p Add space and time complexity.

File: C-Plus-Plus/other/distinct_subsets_of_set.cpp (hunk 2 above)
Comment (somya51p) [reply C-Plus-Plus/other/distinct_subsets_of_set.cpp]: @iamrajiv I have made the suggested changes. Please review the same. Thankyou!

**Comment by somya51p** [comment]
@iamrajiv  I have made the suggested changes. Please review the same. Thankyou!

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@somya51p Please update the readme file.

**Comment by somya51p** [comment]
@iamrajiv Done. Please review the changes, Thankyou!

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**